FirebaseRemoteConfig/Sources/FIRRemoteConfigComponent.m

```diff
@@ -95,10 +100,13 @@ + (void)load {
 + (NSArray<FIRComponent *> *)componentsToRegister {
   FIRDependency *analyticsDep = [FIRDependency dependencyWithProtocol:@protocol(FIRAnalyticsInterop)
                                                            isRequired:NO];
+  FIRDependency *rcInteropDep =
+      [FIRDependency dependencyWithProtocol:@protocol(FIRRemoteConfigInterop) isRequired:NO];
+
   FIRComponent *rcProvider = [FIRComponent
       componentWithProtocol:@protocol(FIRRemoteConfigProvider)
         instantiationTiming:FIRInstantiationTimingAlwaysEager
-               dependencies:@[ analyticsDep ]
+               dependencies:@[ analyticsDep, rcInteropDep ]
```

Comment:
So I think instead of depending on the interop, we need to create another provider for it. Eg. check out AppCheck's interop:

<img width="988" alt="Screenshot 2023-11-22 at 11 25 53 AM" src="https://github.com/firebase/firebase-ios-sdk/assets/555046/ae470153-e7b1-4fae-8acb-4da6686dfd39">

Reply:
So RC will have 2 providers, one for the overall SDK, and one for the Interop... I think

Reply:
RC won't depend on the interop in here, because RC will BE the interop

Reply:
This sounds right to me, and similar to what we currently have in the design:

* The existing `FIRRemoteConfigProvider` will conform to a new `FIRRemoteConfigInterop`. (This might be a distinction, rather than two "sibling" providers)
* The component implements the interop's methods (since it already conforms to `FIRRemoteConfigProvider`).
* Since the component is already aware of the different namespaces, it can pick the right namespace (i.e. instance of RemoteConfig) with which to register the subscriber. 
* That RemoteConfig instance then calls the subscriber(s) as necessary.